warehouse/packaging/utils.py

```diff
@@ -59,32 +62,41 @@ def _simple_detail(project, request):
         "versions": versions,
         "files": [
             {
-                "filename": file.filename,
-                "url": request.route_url("packaging.file", path=file.path),
-                "hashes": {
-                    "sha256": file.sha256_digest,
+                **{
```

Comment:
This is rather hacky but was done to prevent generating a key `provenance` with a `None` value as the PEP740 states : 

> The value of the provenance key SHALL be a JSON string, which SHALL be the SHA-256 digest of the associated .provenance file, as in the Simple Index.

Reply:
We might want to rethink that, I think there is a difference between provenance not being present and provenance not being supported by the API and it might be a good idea to make the distinction (would also make this cleaner).

Reply:
Yeah, true -- my original thinking was that the `API_VERSION` can signal that, but that might be overly constrictive/burdensome, especially if newer versions come out and a particular index still doesn't want to support provenance. I can tweak the PEP to allow `provenance: None`, effectively signaling that the index knows about provenance but isn't presenting any for this project.

Reply:
That makes sense to me and brings this more inline with how we present other data in the API as well!

Reply:
https://github.com/python/peps/pull/3906

Reply:
Much cleaner - thanks for the PEP update :)